tests/foreman/performance/common/perf_candlepin.py

```diff
@@ -0,0 +1,116 @@
+
+        # note: must create ssh keys for vm if running on local
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        result = ssh.command('time -p subscription-manager register \
+                             --activationkey={0} --org={1}'
+                             .format(ak_name, default_org), hostname=vm)
+        if result.return_code != 0:
+            cls.logger.error(result.stderr)
+        else:
+            real_time = [real for real in result.stderr.split('\n')
+                         if real.startswith('real')]
+            real_time = real_time[0].split(' ')[1]
+            return float(real_time)
+
+    @classmethod
+    def single_register_attach(cls, sub_id, default_org, environment, vm):
+        """Subscribe VM to Satellite by Register + Attach"""
+
+        ssh.command(
+            'subscription-manager clean',
+            hostname=vm
+        )
+        time_reg = cls.sub_mgr_register_authentication(default_org,
+                                                       environment, vm)
+        # pool_id = cls.get_subscription_pool_id()
+        time_att = cls.sub_mgr_attach(sub_id, vm)
+
+        if time_reg is None or time_att is None:
+            total_time = 0
+        else:
+            total_time = time_reg + time_att
+        return time_reg, time_att, total_time
+
+    @classmethod
+    def sub_mgr_register_authentication(cls, default_org, environment, vm):
+        """subscription-manager register -u -p --org --environment"""
+        result = ssh.command('time -p subscription-manager \
+                             register --username={0} --password={1} --org={2} \
+                             --environment={3}'
+                             .format(cls.username, cls.password,
+                                     default_org, environment),
+                             hostname=vm)
+
+        if result.return_code == 0:
+            real_time = [real for real in result.stderr.split('\n')
+                         if real.startswith('real')]
+            real_time = real_time[0].split(' ')[1]
+            return float(real_time)
+        else:
+            cls.logger.error(result.stderr)
+
+    @classmethod
+    def sub_mgr_attach(cls, pool_id, vm):
+        """subscription-manager attach --pool=pool_id"""
+
+        result = ssh.command('time -p subscription-manager \
+                             attach --pool={}'
+                             .format(pool_id), hostname=vm)
+
+        if result.return_code == 0:
+            real_time = [real for real in result.stderr.split('\n')
+                         if real.startswith('real')]
+            real_time = real_time[0].split(' ')[1]
+            return float(real_time)
+
+    @classmethod
+    def single_delete(cls, uuid):
+        """Delete system from subscription"""
+
+        start = time.time()
+        r = requests.delete('https://{0}/katello/api/systems/{1}'
```

Comment:
Also is better to use a more descriptive name, like response or res. Single letter names will require others to check what you mean when reading the code.